frontend/public/components/search-filter-dropdown.tsx

```diff
@@ -16,82 +20,84 @@ export enum searchFilterValues {
-      component="button"
-    >
-      {t(searchFilterValues.Name)}
-    </DropdownItemDeprecated>,
-  ];
   const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
     if (e.key === 'Enter') {
       const { value } = e.target as HTMLInputElement;
       onChange(selected, value, true);
     }
   };
-
   const handleInputValue = (_event, value: string) => {
     onChange(selected, value, false);
   };
 
+  const selectItems = [
+    <SelectOption key="label-action" data-test="label-filter" value={searchFilterValues.Label}>
+      {t(searchFilterValues.Label)}
+    </SelectOption>,
+    <SelectOption key="name-action" data-test="name-filter" value={searchFilterValues.Name}>
+      {t(searchFilterValues.Name)}
+    </SelectOption>,
+  ];
+
   return (
-    <div className="pf-v5-c-input-group">
-      <DropdownDeprecated
-        onSelect={onSelect}
-        toggle={
-          <DropdownToggleDeprecated
-            id="toggle-id"
-            onToggle={onToggle}
-            toggleIndicator={CaretDownIcon}
+    <>
+      <InputGroup>
+        <InputGroupItem>
+          <Select
+            isOpen={isOpen}
+            selected={selected}
+            onSelect={onSelect}
+            onOpenChange={(lala) => setOpen(lala)}
+            toggle={(toggleRef: React.Ref<MenuToggleElement>) => {
+              return (
+                <MenuToggle
+                  ref={toggleRef}
+                  isExpanded={isOpen}
+                  onClick={onToggle}
+                  icon={<FilterIcon />}
+                  id="toggle-id"
+                >
+                  {selected}
+                </MenuToggle>
+              );
+            }}
+            shouldFocusToggleOnSelect
           >
-            <>
-              <FilterIcon className="span--icon__right-margin" /> {t(selected)}
-            </>
-          </DropdownToggleDeprecated>
-        }
-        isOpen={isOpen}
-        dropdownItems={dropdownItems}
-      />
-      <TextFilter
-        parentClassName="co-search__filter-input"
-        onChange={handleInputValue}
-        placeholder={selected === searchFilterValues.Label ? 'app=frontend' : 'my-resource'}
-        name="search-filter-input"
-        id="search-filter-input"
-        value={selected === searchFilterValues.Label ? labelFilterInput : nameFilterInput}
-        onKeyDown={handleKeyDown}
-        aria-labelledby="toggle-id"
-      />
-    </div>
+            <SelectList>{selectItems}</SelectList>
+          </Select>
+        </InputGroupItem>
+        <InputGroupItem>
+          <TextFilter
+            onChange={handleInputValue}
+            placeholder={selected === searchFilterValues.Label ? 'app=frontend' : 'my-resource'}
+            name="search-filter-input"
+            id="search-filter-input"
+            value={selected === searchFilterValues.Label ? labelFilterInput : nameFilterInput}
+            onKeyDown={handleKeyDown}
+            aria-labelledby="toggle-id"
```

Comment:
you might make these ids unique (e.g., through using [_.uniqueId](https://lodash.com/docs/4.17.15#uniqueId)) as there could be a problem with this aria label being invalid due to multiple `#toggle-id`s present in the dom

Reply:
Good suggestion, @logonoff.  Updated.  

`toggle-id` was the existing value, but a quick search shows [we're using that same value elsewhere in the codebase](https://github.com/search?q=repo%3Aopenshift%2Fconsole+id%3D%22toggle-id%22&type=code).